mutable_tree.go

```diff
@@ -200,10 +208,12 @@ func (tree *MutableTree) recursiveRemove(node *Node, key []byte) ([]byte, *Node,
 
 		if len(orphaned) == 0 {
 			return node.hash, node, nil, value, orphaned
-		} else if newLeftHash == nil && newLeftNode == nil { // left node held value, was removed
-			return node.rightHash, node.rightNode, node.key, value, orphaned
 		}
 		orphaned = append(orphaned, node)
+		if newLeftHash == nil && newLeftNode == nil { // left node held value, was removed
+			orphaned = append(orphaned, node)
```

Comment:
Is this line duplicated?

Reply:
It seems it is from the right node below
```suggestion
```

Reply:
correct! thanks for the catch